packages/devtools_app/lib/src/performance/performance_controller.dart

```diff
@@ -356,14 +517,37 @@ class PerformanceController
 
   void addTimelineEvent(TimelineEvent event) {
     data.addTimelineEvent(event);
+    _timelineEvents.value.add(event);
```

Comment:
Adding elements to a ValueListenable without notifying isn't safe. Can you use ListValueNotifier or change the _timelineEvents so they are not a ValueListenable?

Reply:
turns out we weren't even using the _timelineEvents notifier in the UI. removed it completely.